bokehjs/src/lib/models/axes/axis.ts

```diff
@@ -179,17 +199,40 @@ export class AxisView extends GuideRendererView {
 
     const [nx, ny] = this.normals
     const standoff = extents.tick + extents.tick_label + this.model.axis_label_standoff
-
     const {vertical_align, align} = this.panel.get_label_text_heuristics("parallel")
-    axis_label.position = {
+    const angle = this.panel.get_label_angle_heuristic("parallel")
+
+    const position = {
       sx: sx + nx*standoff,
       sy: sy + ny*standoff,
       x_anchor: align,
       y_anchor: vertical_align,
     }
-    axis_label.align = align
 
-    axis_label.paint(ctx)
+    const axis_label = text instanceof MathText
+      ? this.axis_label_math_text_view
+      : new TextBox({text})
+
+    axis_label.angle = angle
+    axis_label.visuals = this.visuals.axis_label_text
+    axis_label.position = position
+
+    if (axis_label instanceof MathTextView) {
+      // If the image is not loaded
+      if (!axis_label.svg_image) {
+        // Request a new render
+        this.request_paint()
```

Comment:
Requesting paint here doesn't make sense. Worst case scenario, the request is dispatched immediately and nothing happens when `load_image()` is done, until another (possibly unrelated) paint request is made. This line should be removed and then paint should be requested when `load_image()` has actually anything to render.